internal/controller/serving/reconcilers/llm_role_reconciler.go

```diff
@@ -0,0 +1,138 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/kmeta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/comparators"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/processors"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/resources"
+)
+
+var _ LLMSubResourceReconciler = (*LLMRoleReconciler)(nil)
+
+type LLMRoleReconciler struct {
+	LLMNoResourceRemoval
+	client         client.Client
+	roleHandler    resources.RoleHandler
+	deltaProcessor processors.DeltaProcessor
+}
+
+func NewLLMRoleReconciler(client client.Client) *LLMRoleReconciler {
+	return &LLMRoleReconciler{
+		client:         client,
+		roleHandler:    resources.NewRoleHandler(client),
+		deltaProcessor: processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *LLMRoleReconciler) Reconcile(ctx context.Context, log logr.Logger, llmisvc *kservev1alpha1.LLMInferenceService) error {
+	log.V(1).Info("Verifying that the model user role exists")
+
+	// Create Desired resource
```

Comment:
```suggestion
```

Somewhat obvious by reading next statement :) Can we remove these redundant comments?